Automattic/wp-calypso
# Move localizeUrl to new i18n-utils package

#### Changes proposed in this Pull Request
* Moves lib/i18n-utils' `localizeUrl` function to a new package in `packages/i18n-utils`

#### Testing instructions

localizeUrl tests were included in the move:

```
yarn test-client client/lib/i18n-utils
yarn workspace @automattic/i18n-utils test
```

Manual testing involves checking `localizeUrl` usage still works as expected for example:

- Set locale to german or french using the account settings page http://calypso.localhost:3000/me/account
- Navigate to http://calypso.localhost:3000/me/privacy
- The link to `https://automattic.com/privacy` should be localized (e.g. `https://automattic.com/fr/privacy`) but should point to the original english url for any other language.

---

- Partially fixes #37665 - introduces / exposes localizeUrl for use in apps/packages.
- Enables a fix for #45610
